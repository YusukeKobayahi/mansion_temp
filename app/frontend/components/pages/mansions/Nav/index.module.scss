@import "~/styles/variables";
@import "~/styles/utils";

.container {
  background-color: #ffffff;
  position: sticky;
  margin: 0 calc(50% - 50vw);
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
}

.inner {
  padding: 0 calc(50vw - 500px);
  display: flex;
  background-color: #d6c79b44;
}

a.btn {
  flex: 1;
  text-align: center;
  line-height: 50px;
  font-size: 13px;
  color: #746744;
  font-weight: bold;
  position: relative;

  &::after {
    content: "";
    position: absolute;
    bottom: 4px;
    left: 50%;
    transform: translateX(-50%);
    border-top: 4px solid #746744;
    border-right: 3px solid transparent;
    border-left: 3px solid transparent;
  }

  &:hover {
    color: #c9b47a;

    &::after {
      border-top-color: #c9b47a;
    }
  }
}
