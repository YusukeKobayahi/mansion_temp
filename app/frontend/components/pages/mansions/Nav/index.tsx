import styles from "./index.module.scss";
import classnames from "classnames";
import { MemoryRouter } from "react-router-dom";
import { HashLink } from "react-router-hash-link";

const Nav: React.FC = () => {
  return (
    <section className={classnames(styles.container, "pcOnly")}>
      <div className={styles.inner}>
        <MemoryRouter>
          <HashLink smooth to="#used" className={styles.btn}>
            中古販売価格
          </HashLink>
        </MemoryRouter>
        <MemoryRouter>
          <HashLink smooth to="#info" className={styles.btn}>
            物件の情報
          </HashLink>
        </MemoryRouter>
        <MemoryRouter>
          <HashLink smooth to="#price" className={styles.btn}>
            相場価格
          </HashLink>
        </MemoryRouter>
        <MemoryRouter>
          <HashLink smooth to="#local" className={styles.btn}>
            数字で見る地域情報
          </HashLink>
        </MemoryRouter>
        <MemoryRouter>
          <HashLink smooth to="#detail" className={styles.btn}>
            物件概要
          </HashLink>
        </MemoryRouter>
      </div>
    </section>
  );
};

export default Nav;
