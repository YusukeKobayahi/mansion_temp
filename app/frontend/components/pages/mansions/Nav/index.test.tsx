import { render } from "@testing-library/react";
import Nav from "~/components/pages/mansions/Nav";

describe("components/pages/mansions/Nav", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<Nav />);
    expect(asFragment()).toMatchSnapshot();
  });
});
