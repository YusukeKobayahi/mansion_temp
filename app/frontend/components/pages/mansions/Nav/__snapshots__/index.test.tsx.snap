// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/mansions/Nav 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container pcOnly"
  >
    <div
      class="inner"
    >
      <a
        class="btn"
        href="/#used"
      >
        中古販売価格
      </a>
      <a
        class="btn"
        href="/#info"
      >
        物件の情報
      </a>
      <a
        class="btn"
        href="/#price"
      >
        相場価格
      </a>
      <a
        class="btn"
        href="/#local"
      >
        数字で見る地域情報
      </a>
      <a
        class="btn"
        href="/#detail"
      >
        物件概要
      </a>
    </div>
  </section>
</DocumentFragment>
`;
